### ISSUE TITLE
Gather Data Science feedback on union and tuple types

### ISSUE THREAD

**Issue by jklukas** [mozilla/jsonschema-transpiler#88]
## Wanted: Data science review of query interface

We have a few remaining ambiguous types in JSON schemas that are preventing some important ping fields from appearing as fields in BigQuery. Currently, these ambiguous values end up as part of the `additional_properties` JSON blob in BigQuery ping tables, so they are available but awkward and potentially expensive to query.

This issue lays out the proposed interface for presenting these as fields. We want to gather feedback now from users of the data, because deploying these changes will be to some extent irreversible; once we coerce these fields to a certain BQ type, we cannot change our minds and choose a different type for an existing field.

For both of these schema issues, we're going to use the `event` ping to demonstrate.

tl;dr Please play with the query given below in https://github.com/mozilla/jsonschema-transpiler/issues/88#issuecomment-525430214 and leave feedback in this issue about any potential gotchas or improvements you'd like to see with the proposed transformations.

**Comment by sunahsuh** [comment]
>  We can preserve the original JSON types by including quotes in the JSON string if the original value was a string, but then the user would need to strip out the quotes.

Desktop Firefox only allows strings anyway for both the key and value, which we added to the event specification early on due to lack of union type support in Parquet. (See the blurb for "extra" here: https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/collection/events.html#serialization-format)
